server/api/permission/list/controller.js

```diff
@@ -26,17 +29,79 @@ const validation = validateRequest({
     cursor: Joi.string()
       .base64()
       .optional(),
+    scope: Joi.string()
+      .base64()
+      .optional(),
+    role: Joi.string()
+      .base64()
+      .optional(),
+    isSystemPermission: Joi.boolean()
+      .optional(),
   },
 });
 
+const visitRequestedRole = async ({ request, db }, next) => {
+  const roleId = request.body.role;
+  if (roleId) {
+    const role = await getRoleInfo(db, { id: roleId });
+
+    // visit session with requested role data
+    request.session = {
+      ...request.session,
+      role,
+    };
+  }
+  await next();
+};
+
+const isUserAuthorised = async ({ request }, next) => {
+  // verify a user has access to the requested org
+  if (request.body.scope) {
+    const isScopeAccessible = findUserPermissionIndex(request.session.user.permissions, {
+      name: 'yeep.role.read',
+      orgId: request.body.scope,
+    }) !== -1;
+
+    if (!isScopeAccessible) {
+      throw new AuthorizationError(
+        `User "${
+          request.session.user.username
+        }" does not have sufficient permissions to list permissions under org ${request.body.scope}`
```

Comment:
Suggestion: how about we slightly change the copy as follows...

`User${request.session.user.username} is not authorized to list permissions under org ${request.body.scope}`

I mostly suggest this to avoid repetition of the "permission" word.